src/main.rs

```diff
@@ -1,46 +1,34 @@
+mod argument_parser;
+mod linking;
+
 use std::{
     borrow::Cow,
     env,
+    ffi::OsStr,
     fs::{self, File},
     io::Write,
     ops::RangeInclusive,
     path::{Path, PathBuf},
-    process::{self, Command},
+    process,
 };
 
 use anyhow::anyhow;
 use object::{elf, Object as _, ObjectSection, SectionFlags};
 
-const EXIT_CODE_FAILURE: i32 = 1;
-// TODO make this configurable (via command-line flag or similar)
-const LINKER: &str = "rust-lld";
 /// Stack Pointer alignment required by the ARM architecture
 const SP_ALIGN: u64 = 8;
 
 fn main() -> anyhow::Result<()> {
-    notmain().map(|code| process::exit(code))
```

Comment:
Okay. I personally was really confused by this pattern and also think it makes the signature and returns of `fn main` a bit weird.

If you think it makes the code more understandable I am happy to re-add it though.

Reply:
Hrmm! then that's a valid concern. I've kind of assumed that this `main -> notmain` pass-through is a common pattern in (embedded?) Rust but iirc I've also stumbled there earlier. Maybe @japaric has some insights there?

Reply:
To clarify: What confused me is mainly that following line (and similar) is actually an Error, although being `Ok(...)`: https://github.com/knurling-rs/flip-link/blob/3e43babd960777d7808747f89bc7c43604011742/src/main.rs#L36

Reply:
Huh. That's similar to some probe-run parts I also found misleading :D

Reply:
Yes. Whatever we decide here I would like to do the same thing in `probe-run`, just for consistency reasons.

Reply:
I will revert this change, since, as pointed out by @japaric and the `fn process::exit` docs, "no destructors on the current stack or any other thread's stack will be run" when calling it. So it's actually not just aesthetics.